posts/views.py

```diff
@@ -0,0 +1,30 @@
+from django.shortcuts import render, redirect
+from django.utils.decorators import method_decorator
+from django.views import View
+
+from categories.models import Category
+from posts.decorator import user_required
+from posts.forms import PostForm
+
+
+@method_decorator(user_required, name="dispatch")
+class CreatePostView(View):
```

Comment:
method decorator works only for methods
there is a good place to use `LoginRequiredMixin`, I think it has everything u need
```suggestion
class CreatePostView(LoginRequiredMixin, View):
```